include/libdnf5/rpm/transaction_callbacks.hpp

```diff
@@ -69,6 +69,13 @@ class TransactionCallbacks {
 
     virtual ~TransactionCallbacks() = default;
 
+    /// Called right before the rpm transaction is run
+    /// @param total Number of elements in the rpm transaction
+    virtual void before_begin(uint64_t total) {}
```

Comment:
/usr/include/libdnf5/rpm/transaction_callbacks.hpp is a public header. Adding virtual methods breaks its ABI. Can this feature be implemented without virtual methods, or in a private class? Otherwise, we need to postpone this pull request until a window for ABI change opens.

Reply:
Ech. OK then, moving the PR to 5.2.